src/System.Management.Automation/engine/PSConfiguration.cs

```diff
@@ -331,7 +360,23 @@ internal PSKeyword GetLogKeywords()
             return result;
         }
 #endif // UNIX
- 
+
+        /// <summary>
+        /// Read a value from the configuration file.
+        /// </summary>
+        /// <typeparam name="T">The type of the value</typeparam>
+        /// <param name="scope">The ConfigScope of the configuration file to update.</param>
+        /// <param name="key">The string key of the value.</param>
+        /// <param name="defaultValue">The default value to return if the key is not present.</param>
+        /// <param name="readImpl"></param>
+        private T ReadValueFromFile<T>(ConfigScope scope, string key, T defaultValue = default(T),
+                                       Func<JToken, JsonSerializer, T, T> readImpl = null)
+        {
+
+            string fileName = GetConfigFilePath(scope);
+            return ReadValueFromFile<T>(fileName, key, defaultValue, readImpl);
```

Comment:
How about just changing the old `ReadValueFromFile<T>(string fileName ...` to `ReadValueFromFile<T>(ConfigScope scope ...`? Same with the other file operation help methods below. It feels cleaner that way.

Reply:
@daxian-dbw Are you suggesting merging versus wrapping?